### ISSUE TITLE
Composition over Inheritance

### ISSUE THREAD

**Issue by Turbo87** [alexander-alvarez/ember-data-test-helpers#5]
The proposed helpers (`setupStoreTest`, `setupSerializerTest`, ...) are in some sense using inheritance to create new tests since they seem to be calling `setupTest()` under the hood. IMHO it makes more sense to use composition instead:

```js
setupTest(hooks);
setupStore(hooks);

// instead of
setupStoreTest(hooks);
```

This will also make the tests easier to understand since there are fundamentally only three different kinds of tests ("container", rendering, application) with several things built on top of them (e.g. `setupStore`).

**Comment by alexander-alvarez** [comment]
Fair enough. I guess that means then that the approach then to mitigate the onboarding complexity via generators. Because throwing lots of different concepts in your face as a new user IMO would be intimidating.

Nonetheless, I buy it
